omegaconf/omegaconf.py

```diff
@@ -831,9 +855,10 @@ def _maybe_wrap(
     is_optional: bool,
     parent: Optional[BaseContainer],
 ) -> Node:
-    # if already a node, update key and parent and return as is.
-    # NOTE: that this mutate the input node!
-    if isinstance(value, Node):
+    # If already a node, update key and parent and return as is.
+    # NOTE: that this mutates the input node!
+    # If `ref_type` is a union we still need to wrap the node into a `UnionNode`.
+    if isinstance(value, Node) and not _is_union(ref_type):
```

Comment:
I have a concern about this.
can this lead to multiple wrapping in a Union node?

Reply:
So, right now in the test suite `value` is never a `UnionNode` here at least.

That being said, I do have concerns with unions of unions (ex: `Uniont[int, Union[str, float]]`). I suspect some things may not work properly. However it may not be worth supporting them since we can combine them in a single `Union` (maybe we should catch it and raise an error though, to be safe).

Reply:
I think Python is doing it automatically. try that in an interpreter.

Reply:
Oooh nice :) In that case I think it should be ok -- maybe we could add an `assert not isinstance(value, UnionNode)` in the `else:` block to be safe (unless you can see a way to trigger it in a normal use case?)

Reply:
We can block this until we see an actual use case. Better having someone complain about something raising an exception than have internal structure that goes haywire in certain scenarios.